modules/3d-tiles/src/tileset/tile-3d-header.js

```diff
@@ -11,13 +11,55 @@ import {createBoundingVolume} from './helpers/bounding-volume';
 // TODO - inject this dependency?
 import {fetchFile} from '@loaders.gl/core';
 
-const defined = x => x !== undefined;
+const defined = x => x !== undefined && x !== null;
 
 /* eslint-disable */
 const scratchDate = new Date();
 const scratchCommandList = [];
 const scratchToTileCenter = new Vector3();
 
+// TODO: Remove
+const scratchPlane = new Plane();
+import {INTERSECT, Intersect, Plane} from '@math.gl/culling';
+
+function computeVisibilityWithPlaneMask(cullingVolume, boundingVolume, parentPlaneMask) {
+  assert(boundingVolume, 'boundingVolume is required.');
+  assert(Number.isFinite(parentPlaneMask), 'parentPlaneMask is required.');
+
+  if (
+    parentPlaneMask === CullingVolume.MASK_OUTSIDE ||
+    parentPlaneMask === CullingVolume.MASK_INSIDE
+  ) {
+    // parent is completely outside or completely inside, so this child is as well.
+    return parentPlaneMask;
+  }
+
+  // Start with MASK_INSIDE (all zeros) so that after the loop, the return value can be compared with MASK_INSIDE.
+  // (Because if there are fewer than 31 planes, the upper bits wont be changed.)
+  let mask = CullingVolume.MASK_INSIDE;
+
+  const planes = cullingVolume.planes;
+  for (let k = 0; k < cullingVolume.planes.length; ++k) {
+    // For k greater than 31 (since 31 is the maximum number of INSIDE/INTERSECTING bits we can store), skip the optimization.
+    const flag = k < 31 ? 1 << k : 0;
+    if (k < 31 && (parentPlaneMask & flag) === 0) {
+      // boundingVolume is known to be INSIDE this plane.
+      continue;
+    }
+
+    const plane = scratchPlane.fromNormalDistance(planes[k].normal, planes[k].distance);
```

Comment:
@ibgreen Couldn't use the real culling-volume.js function yet since it errors on `fromCoefficients(...planes[k])`. I guess what was actually meant was `fromNormalDistance`. I switched this placeholder function to use it and test it out.

Reply:
Good question, I was a bit uncertain here. I initially wanted to use `fromNormalDistance`.

But as far as I could tell, in other Cesium classes (i.e. the frustum classes), the planes are generated as coefficients (`Cartesian4` = `Vector4`). 

So as a start I tried to stick with the existing conventions until I had tests passing. 

Perhaps I can add a check and support both representations.

Reply:
Couldn't you do a quick conversion on your side, something like `planes.map(({n, d}) => [n.x, n.y, n.z, d]`

Reply:
This would work but there's an over eager assert in the `fromCoefficients` checking if the normal is exactly 1 (it died on 0.9999999999999999...)
`const planesXYZD = planes.map(({normal, distance}) => [normal.x, normal.y, normal.z, distance]);`

`const plane = scratchPlane.fromCoefficients(...planesXYZD[k]);`

cullingVolume is a frameState variable that gets passed around a bunch of places not sure we would want to re-org its data like this. I don't think anything else actually consumes it but still. Or is this experiment more of a test for that function.